### PR TITLE
Add bulk repository deletion functionality (PROJQUAY-4080)

### PR DESCRIPTION
**Pull request by harishsurf**
This PR covers the following Jira:
- https://issues.redhat.com/browse/PROJQUAY-4079
- https://issues.redhat.com/browse/PROJQUAY-4080  - fixed for RepoList page. still seeing issues for OrgList page (will keep the jira open and follow up)
- https://issues.redhat.com/browse/PROJQUAY-4091 and https://issues.redhat.com/browse/PROJQUAY-3898 related to Bulk deletion of repo

Screen Shot 2022-08-03 at 5 17 14 PM

ToDo:
- add the generic pagination component to the bulk delete modal
- Fix state management for orgList page
- Fix delete org 

Signed-off-by: harishsurf <hgovinda@redhat.com>

### CODE REVIEW

File: src/routes/RepositoriesList/RepositoriesList.tsx [hunk 1]
```diff
@@ -408,15 +437,14 @@ export default function RepositoriesList() {
...
                   <Td dataLabel={columnNames.visibility}>
                     {' '}
-                    {repo.isPublic ? 'public' : 'private'}
+                    {repo.is_public ? 'public' : 'private'}
                   </Td>
-                  <Td dataLabel={columnNames.tags}> {repo.tags} </Td>
-                  <Td dataLabel={columnNames.size}> {repo.size} </Td>
-                  <Td dataLabel={columnNames.pulls}> {repo.pulls} </Td>
-                  <Td dataLabel={columnNames.lastPull}> {repo.lastPull} </Td>
+                  <Td dataLabel={columnNames.tags}> dummy </Td>
```
Comment (HammerMeetNail) [inline src/routes/RepositoriesList/RepositoriesList.tsx:442]: Do these need to be dummy?

File: src/resources/RepositoryResource.ts [hunk 2]
```diff
@@ -79,3 +78,18 @@ export async function setRepositoryVisibility(
...
+    const response = await Promise.all(
+      repos.map((repo) => {
+        return axios.delete(
+          `/api/v1/repository/${repo.namespace + '/' + repo.name}`,
+        );
+      }),
+    );
+    return response;
+  } catch (error) {
+    console.log(error);
```
Comment (HammerMeetNail) [inline src/resources/RepositoryResource.ts:93]: Can any console.logs like this be converted to console.error?

**Review by bcaton85 — CHANGES_REQUESTED** [review]
Hi Harish, overall looks good. Walking through I noted some issues. You mentioned some will be resolved in a future PR but i'm noting them here for documentation purposes. Not all are intended to be resolved in this PR but JIRA's should be created for those that aren't.

- Org list
    - Organization can be created successfully but will only display after refreshing page
    - Creating multiple org’s duplicates the list
        - eg. If the list shows repo1, repo2 and I create repo3. The list now shows repo1, repo2, repo1, repo2.
    - I was able to create an organization with the email `test`, do we want to check formatting?
    - Deleting organizations deleted immediately - no confirmation given.
    - My user organization doesn’t show up, is that by design?
    - Is the trash icon going to be replaced with a kebab in a future PR? I believe @Sunandadadi ’s work on a generic toolbar may resolve this.
- Repository list for single Org
    - If list is empty the loading sign spins indefinitely. Needs “No repo’s found display”
- Repository list
    - Can’t de-select individual rows
    - Can’t create repo in user organization, is this by design?
    - Creating a repo intermittently returns a 400. Doesn’t seem to pass the namespace. Here is the response from the network tab.
        ```bash
        detail: "None is not of type 'string'\n\nFailed validating 'type' in schema['properties']['namespace']:\n    {'description': 'Namespace in which the repository should be created. '\n                    'If omitted, the username of the caller is used',\n     'type': 'string'}\n\nOn instance['namespace']:\n    None"
        error_message: "None is not of type 'string'\n\nFailed validating 'type' in schema['properties']['namespace']:\n    {'description': 'Namespace in which the repository should be created. '\n                    'If omitted, the username of the caller is used',\n     'type': 'string'}\n\nOn instance['namespace']:\n    None"
        error_type: "invalid_request"
        status: 400
        title: "invalid_request"
        type: "http://localhost/api/v1/error/invalid_request"
        ```
    - If list is empty the loading sign spins indefinitely. Needs “No repo’s found display”
    - State is shared between repo list and repo list for a single org. If I select a repo in Repository list, then go to the same repo through the Organization list, the same rows are selected.

Let me know your thoughts.

File: src/routes/RepositoriesList/RepositoriesList.tsx (hunk 1 above)
Comment (harishsurf) [reply src/routes/RepositoriesList/RepositoriesList.tsx]: replaced with `-` for now. These will be soon addressed as part of https://issues.redhat.com/browse/PROJQUAY-4082

**Comment by HammerMeetNail** [comment]
@bcaton85 Anything empty state related will probably be covered by @jonathankingfc in https://issues.redhat.com/browse/PROJQUAY-4109

**Comment by harishsurf** [comment]
@bcaton85 , thanks for testing it out. Following up on each item: 
- Org list
    - Organization can be created successfully but will only display after refreshing page  - **fixed**
    - Creating multiple org’s duplicates the list. - **fixed** 
    - I was able to create an organization with the email `test`, do we want to check formatting?  - **added a basic email regex validation**
    - Deleting organizations deleted immediately - **fixed**
    - My user organization doesn’t show up, is that by design? - **good catch! the `user org` isn't directly provided when fetching `orgs`, added an extra API call to fetch the user org**
    - Is the trash icon going to be replaced with a kebab in a future PR? I believe @Sunandadadi ’s work on a generic toolbar may resolve this. - **replaced with kebab option**
- Repository list for single Org
    - If list is empty the loading sign spins indefinitely. Needs “No repo’s found display”.   - **there is already [PROJQUAY-4109](https://issues.redhat.com/browse/PROJQUAY-4109) open for this**
- Repository list
    - Can’t de-select individual rows - should be fixed when @Sunandadadi implements the generic select checkbox
    - Can’t create repo in user organization, is this by design?  - **[PROJQUAY-3900](https://issues.redhat.com/browse/PROJQUAY-3900) is open for this**
    - Creating a repo intermittently returns a 400. Doesn’t seem to pass the namespace. Here is the response from the network tab.  - **should be fixed now**
        ```bash
        detail: "None is not of type 'string'\n\nFailed validating 'type' in schema['properties']['namespace']:\n    {'description': 'Namespace in which the repository should be created. '\n                    'If omitted, the username of the caller is used',\n     'type': 'string'}\n\nOn instance['namespace']:\n    None"
        error_message: "None is not of type 'string'\n\nFailed validating 'type' in schema['properties']['namespace']:\n    {'description': 'Namespace in which the repository should be created. '\n                    'If omitted, the username of the caller is used',\n     'type': 'string'}\n\nOn instance['namespace']:\n    None"
        error_type: "invalid_request"
        status: 400
        title: "invalid_request"
        type: "http://localhost/api/v1/error/invalid_request"
        ```
    - If list is empty the loading sign spins indefinitely. Needs “No repo’s found display” - [PROJQUAY-4109](https://issues.redhat.com/browse/PROJQUAY-4109) 
    - State is shared between repo list and repo list for a single org. If I select a repo in Repository list, then go to the same repo through the Organization list, the same rows are selected. - **created [PROJQUAY-4265]**(https://issues.redhat.com/browse/PROJQUAY-4265)

Could you PTAL?

**Comment by HammerMeetNail** [comment]
@harishsurf This looks good, validated most of your fixes. Email validation is kinda simple but probably good enough for now. Can you update the create org modal so that if validation fails the user can't submit and create the org?

**Comment by harishsurf** [comment]
@HammerMeetNail , added validation for create org and fixed regex, PTAL

**Review by HammerMeetNail — APPROVED** [review]
LGTM

**Pull request merged**